Kudu.Core/Deployment/Generator/ExternalCommandBuilder.cs

```diff
@@ -45,9 +54,29 @@ protected ExternalCommandBuilder(IEnvironment environment, IDeploymentSettingsMa
 
         public abstract Task Build(DeploymentContext context);
 
-        protected void RunCommand(DeploymentContext context, string command)
+        public void PostBuild(DeploymentContext context)
+        {
+            context.Logger.Log("Running post deployment command(s)...");
+
+            foreach (var file in this.GetPostBuildActionScripts())
+            {
+                var fi = new FileInfo(file);
+                if (string.Equals(".ps1", fi.Extension))
+                {
+                    string scriptFilePath = string.Format(CultureInfo.InvariantCulture, "PowerShell.exe -ExecutionPolicy RemoteSigned -File \"{0}\"", file);
+                    RunCommand(context, scriptFilePath, fi.Name.Replace(fi.Extension, string.Empty));
```

Comment:
Also, what you have would I think not be correct for names like Hello.foo.foo, as it would remove more than the extension. GetFileNameWithoutExtension does the right thing.